### ISSUE TITLE
Voice connection ready event not firing

### ISSUE THREAD

**Issue by xenorio** [abalabahaha/eris#1436]
I've got a very simple setup for playing some audio. Goes like this:

```js
let connection

async function join() {
    connection = await channel.join({ // I'm leaving out where it's getting the channel from
        selfDeaf: true
    })
    connection.on('ready', () => {
        play()
    })
    connection.on('end', () => {
        play()
    })
}

async function play() {
    if(!connection)return join()
    // Insert starting the stream etc
}
```

According to docs, the "ready" event should fire once the connection becomes ready: https://abal.moe/Eris/docs/0.17.1/VoiceConnection#event-ready

However, it just never fires. Which means it never gets to the point of starting the stream. The bot does connect to the channel, and with some debugging I've even found out the "ready" **property** becomes true, but the event is just sitting there untouched. The "end" event does work properly (when not waiting for ready), so this can't be just me being stupid.

Without waiting for the connection to become ready, I'm sometimes running into `Error: Not ready yet`

Using version 0.17.1

**Comment by xenorio** [comment]
**Exact** code this is based on: https://github.com/ResearchCommunity/Alfred/blob/bb79596c14a7df501b5bb4402ef73c6d3eddd44e/src/music/player.js

**Comment by abalabahaha** [comment]
Docs could probably be clearer, but `join()` waits for the `ready` event.

**Comment by xenorio** [comment]
How exactly is it possible that I sometimes get the "not ready yet" error then?

**Comment by AdamJessop** [comment]
I am seeing this same issue recently. The bot is joining the voice channel, however the promise is never resolving. When I try to bypass the promise and manually resolve it or mark it as void and continue to try and play audio, I get an error that the connection is not ready yet.

**Comment by abalabahaha** [comment]
@AdamJessop that sounds like the recent Discord voice change fixed in 0.17.2, which is unrelated to this issue

**Comment by AdamJessop** [comment]
Thanks @abalabahaha I believe you are right, I thought I had updated the library and turns out it hadn't done so at all. All working with 0.17.2, much appreciated.
